lib/QubitInformationObjectXmlCache.class.php

```diff
@@ -0,0 +1,229 @@
+    {
+      mkdir($eadExportPath, 0755, true);
+    }
+
+    $dcExportPath = $exportsPath . DIRECTORY_SEPARATOR .'dc';
+    if (!is_dir($dcExportPath))
+    {
+      mkdir($dcExportPath, 0755, true);
+    }
+  }
+
+  /**
+   * Export information object to EAD (if top-level) and DC.
+   *
+   * @param object  information object to be export
+   *
+   * @return null
+   */
+  public function export($resource)
+  {
+    // Only cache top-level information object's EAD XML
+    if ($resource->parentId == QubitInformationObject::ROOT_ID)
+    {
+      $this->cacheXmlRepresentation($resource, 'ead');
+    }
+
+    $this->cacheXmlRepresentation($resource, 'dc');
+  }
+
+  /**
+   * Export all information objects to EAD (if top-level) and DC.
+   *
+   * @return null
+   */
+  public function exportAll()
+  {
+    foreach (QubitInformationObject::getAll() as $io)
+    {
+      $published = $io->getPublicationStatus()->statusId == QubitTerm::PUBLICATION_STATUS_PUBLISHED_ID;
+
+      // Only cache if not root and published
+      if ($io->id != QubitInformationObject::ROOT_ID && $published)
+      {
+        $this->export($io);
+      }
+    }
+  }
+
+  /**
+   * Export information object to EAD (if top-level) and DC.
+   *
+   * @param object  information object to cache
+   * @param string  format of XML ("dc" or "ead")
+   *
+   * @return null
+   */
+  protected function cacheXmlRepresentation($resource, $format)
+  {
+    $tempFile = tmpfile();
+    fwrite($tempFile, $this->generateXmlRepresentation($resource, $format));
+    $metadata = stream_get_meta_data($tempFile);
+    $this->storeXmlExport($metadata['uri'], $resource->id, $format);
+    fclose($tempFile);
+  }
+
+  /**
+   * Generate XML representation of information object.
+   *
+   * @param object  information object to cache
+   * @param string  format of XML ("dc" or "ead")
+   *
+   * @return string  XML representation
+   */
+  protected function generateXmlRepresentation($resource, $format)
+  {
+    exportBulkBaseTask::includeXmlExportClassesAndHelpers();
+    $rawXml = exportBulkBaseTask::captureResourceExportTemplateOutput($resource, $format);
+    return Qubit::tidyXml($rawXml);
+  }
+
+  /**
+   * Store XML representations of information object.
+   *
+   * Two XML files are created, copied from a source file to files within the
+   * downloads directory. One copy contains the complete XML while the other
+   * removes the XML declaration line (and, for EAD, the doctype line).
+   *
+   * The copy with the line(s) removed is created so it can be included within
+   * OAI-PMH results.
+   *
+   * @param string  path to temporary file containing XML
+   * @param integer  ID of information object to cache
+   * @param string  format of XML ("dc" or "ead")
+   *
+   * @return null
+   */
+  protected function storeXmlExport($filePath, $objectId, $format)
+  {
+    // Copy unmodified XML to downloads subdirectory
+    copy($filePath, $this->getFilePath($objectId, $format));
```

Comment:
Is this ok? Can you invoke `getFilePath` like that when it is static? Sorry my PHP is rusty.